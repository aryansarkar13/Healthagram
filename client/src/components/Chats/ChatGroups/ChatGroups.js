import React, { Component } from "react";
import ChatGroup from "components/Chats/ChatGroups/ChatGroup/ChatGroup";
import { connect } from "react-redux";
import CreateChatGroup from "components/Chats/ChatGroups/CreateChatGroup/CreateChatGroup";

export class ChatGroups extends Component {
  filterByCountry = groups => {
    const filterCountry = this.props.filterCountry;
    const filterText = this.props.filterText;

    //If there are no inputs on country and text field of filter, just return
    if (filterCountry === "" && filterText === "") {
      return groups;
    }

    return groups.filter(group => {
      const textMatch =
        filterText === "" ||
        group.name.toLowerCase().includes(filterText.toLowerCase());
      const countryMatch =
        filterCountry === "" || group.country === filterCountry;
      return textMatch && countryMatch;
    });
  };
  renderChatGroups = () => {
    let render = null;
    //cloudinary image link with setting

    if (this.props.groups) {
      //Filters the group list by country
      const filteredGroups = this.filterByCountry(this.props.groups);
      if (filteredGroups.length === 0) {
        return <p>There are no groups to display </p>;
      }
      render = filteredGroups.map((group, i) => {
        const editable = this.props.profile.id === group.createdBy;
        console.log("The group is editable", editable);
        return (
          <div className="col-md-3 col-sm-6">
            <ChatGroup
              name={group.name}
              country={group.country}
              image={group.image}
              id={group._id}
              key={group._id}
              favourites={group.favourites.length}
              editable={editable}
            />
          </div>
        );
      });
    }
    return render;
  };
  render() {
    return (
      <div className="ChatGroups">
        <div className="container ChatGroups__container">
          <div className="row">
            <div className="col-md-3 col-sm-6 ChatGroups__CreateChatGroup__section">
              <CreateChatGroup />
            </div>
            {this.renderChatGroups()}
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  profile: state.profile,
  groups: state.group.lists,
  filterCountry: state.group.filter.country,
  filterText: state.group.filter.text
});

export default connect(mapStateToProps)(ChatGroups);
