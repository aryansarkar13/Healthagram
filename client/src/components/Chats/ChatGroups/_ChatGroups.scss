@import "CreateChatGroup/CreateChatGroup";
@import "ChatGroup/ChatGroup";

.ChatGroups__CreateChatGroup__section {
  @include for-phone-only {
    display: none;
  }
}

.ChatGroups__container {
  .col-md-3 {
    @media (min-width: 576px) and(max-width: 1200px) {
      max-width: 50% !important;
      flex: 0 0 50% !important;
    }
  }
}
