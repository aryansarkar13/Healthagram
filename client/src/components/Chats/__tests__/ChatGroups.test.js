import { ChatGroups } from "components/Home/ChatGroups/ChatGroups";
import React from "react";
import { shallow } from "enzyme";
describe("The ChatGroups component", () => {
  it("should not regress", () => {
    const groups = [
      {
        name: "Samrat",
        country: "Nepal",
        image: "randomstring"
      }
    ];
    const wrapper = shallow(<ChatGroups groups={groups} />);
    expect(wrapper).toMatchSnapshot();
  });
});
