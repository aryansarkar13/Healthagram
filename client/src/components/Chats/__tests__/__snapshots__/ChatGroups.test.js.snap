// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ChatGroups component should not regress 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatGroups
    groups={
        Array [
            Object {
              "country": "Nepal",
              "image": "randomstring",
              "name": "Samrat",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(ChatGroup)
          country="Nepal"
          image="https://res.cloudinary.com/samrat/image/upload/c_fill,g_face,h_100,w_106/v1540572400/randomstring"
          name="Samrat"
/>,
      ],
      "className": "ChatGroups",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "country": "Nepal",
          "image": "https://res.cloudinary.com/samrat/image/upload/c_fill,g_face,h_100,w_106/v1540572400/randomstring",
          "name": "Samrat",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(ChatGroup)
            country="Nepal"
            image="https://res.cloudinary.com/samrat/image/upload/c_fill,g_face,h_100,w_106/v1540572400/randomstring"
            name="Samrat"
/>,
        ],
        "className": "ChatGroups",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "country": "Nepal",
            "image": "https://res.cloudinary.com/samrat/image/upload/c_fill,g_face,h_100,w_106/v1540572400/randomstring",
            "name": "Samrat",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
